### PR TITLE
Add external variables support for Python bindings

### PR DESCRIPTION
**Pull request by lebauce**
This PR allows setting Python callbacks for attribute pre-update and external variables setting

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

**Comment by lebauce** [comment]
@yunhanw-google Thank you for your review. I added

**Comment by lebauce** [comment]
Could someone please have an other look ? Thanks!

**Comment by lebauce** [comment]
I don't what to do to make this PR move forward :-( Could @bzbarsky-apple @andy31415 please help on this ? Thank you

**Comment by bzbarsky-apple** [comment]
Needs review from someone familiar with the Python bindings bits...  @cecille @yunhanw-google @yufengwangca do you know who might be able to take a look?

File: src/controller/python/chip/server/ServerInit.cpp [hunk 1]
```diff
@@ -99,23 +100,40 @@ void CleanShutdown()
 using PostAttributeChangeCallback = void (*)(EndpointId endpoint, ClusterId clusterId, AttributeId attributeId, uint8_t type,
                                              uint16_t size, uint8_t * value);
 
+using PreAttributeChangeCallback = chip::Protocols::InteractionModel::Status (*)(EndpointId endpoint, ClusterId clusterId,
```
Comment (andy31415) [inline src/controller/python/chip/server/ServerInit.cpp:103]: how do we handle different status codes (IM::Status vs EmberAfStatus) returned from Python? Should we have constants for them or are such constants already defined?

Could we have that documented somewhere on what common statuses mean to be returned from the callback?

**Comment by woody-apple** [comment]
@lebauce ?

**Comment by andy31415** [comment]
No updates for 9 months, assuming stale. Will close.

**Pull request closed without merging**